lib/matplotlib/ticker.py

```diff
@@ -2421,17 +2429,16 @@ def tick_values(self, vmin, vmax):
                 raise ValueError(
                     "Data has no positive values, and therefore cannot be log-scaled.")
 
-        _log.debug('vmin %s vmax %s', vmin, vmax)
-
         if vmax < vmin:
             vmin, vmax = vmax, vmin
-        log_vmin = math.log(vmin) / math.log(b)
-        log_vmax = math.log(vmax) / math.log(b)
-
-        numdec = math.floor(log_vmax) - math.ceil(log_vmin)
+        # Min and max exponents -- float and int versions.
```

Comment:
This schematic helped me a lot in understanding what is going on

```suggestion
        # Min and max exponents -- float and int versions.
        #         ┌── efmax
        # b^(n+2) ┤   emax = n+2
        #         │
        # b^(n+1) ┤	           nd = emax - emin + 1 = 3
        #         │
        #     b^n ┤   emin = n    
        #         └── efmin
```